packages/millicast-sdk/src/utils/Diagnostics.js

```diff
@@ -10,6 +11,36 @@ let _feedId = ''
 let _connection = ''
 const _stats = []
 
+function transformWebRTCStatsToCMCD (diagnostics) {
+  // Helper function to map individual stat objects to CMCD-like structure
+  function mapStats (type, stat) {
+    return {
+      ts: Math.round(stat.timestamp) || '', // Timestamp to the nearest millisecond
+      ot: type === 'audio' ? 'a' : 'v', // 'a' for audio, 'v' for video
+      bl: stat.jitterBufferDelay || 0, // Buffer length from jitterBufferDelay, default to 0 if not available
+      br: Math.round(stat.bitrate || 0), // Bitrate, rounded to nearest integer, default to 0 if not available
+      pld: stat.packetsLostDeltaPerSecond || 0, // Packets lost delta per second, default to 0 if not available
+      j: stat.jitter || 0, // Jitter, default to 0 if not available
+      mtp: stat.packetRate || 0, // Measured throughput, approximated by packet rate, default to 0 if not available
+      mid: stat.mid || '', // Media ID or track identifier, default to empty string if not available
+      mimeType: stat.mimeType || '' // MIME type of the media stream, default to empty string if not available
+    }
+  }
+
+  diagnostics.stats = diagnostics.stats.reduce((acc, stat) => {
+    const audioStats = stat.audio.inbounds.length !== 0
+      ? stat.audio.inbounds.map(statAudio => mapStats('audio', statAudio))
+      : stat.audio.outbounds.map(statAudio => mapStats('audio', statAudio))
+    const videoStats = stat.video.inbounds.length !== 0
+      ? stat.video.inbounds.map(statVideo => mapStats('video', statVideo))
+      : stat.video.outbounds.map(statVideo => mapStats('video', statVideo))
```

Comment:
@dubeyShivank In refactored this lines as they were only considering inbound stats, so when testing publish side CMCD stats where not shown. Please review it as well.

Reply:
Thanks Santi, good call out. I will address it